src/Domain/BlockNeighbor.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/BlockNeighbor.hpp"
+
+#include <algorithm>
+#include <ostream>
+
+template <size_t VolumeDim>
+BlockNeighbor<VolumeDim>::BlockNeighbor(
+    const size_t id, const Orientation<VolumeDim>& orientation)
+    : id_(id), orientation_ptr_{orientation.get_clone()} {}
```

Comment:
Point for discussion: Should functions like this take a unique_ptr so that they don't have to clone?